src/core/server/rendering/rendering_service.tsx

```diff
@@ -100,6 +100,15 @@ export class RenderingService {
           opensearchDashboardsConfig as OpenSearchDashboardsConfigType
         );
 
+        let locale = i18n.getLocale();
```

Comment:
Is this always a defaulted/configured valid locale?

Reply:
defaulted: yes
valid: no

i18n starts with `en` as the default locale and if configured in the YAML config, it uses that without validating that it has been registered or not; I think this is because while reading the config, the filesystem might not have been read to prime the translation registration cache. At runtime, the locale is validation and if found to not have been registered, `en` is used.

Reply:
From the code, if invalid, then does it potentially lead to a translation file not being loaded? Is this okay because english translations are loaded worst case?

Reply:
If the locale is not registered, it would just use English, yes.